fenix/app/src/main/java/org/mozilla/fenix/browser/TranslationsBinding.kt

```diff
@@ -36,37 +38,63 @@ class TranslationsBinding(
 ) : AbstractBinding<BrowserState>(browserStore) {
 
     override suspend fun onState(flow: Flow<BrowserState>) {
-        flow.mapNotNull { state -> state.findTab(sessionId) }.distinctUntilChangedBy {
-            it.translationsState
-        }.collect { sessionState ->
-            val translationsState = sessionState.translationsState
+        // Browser level flows
+        val browserFlow = flow.mapNotNull { state -> state }
```

Comment:
Updated, to `.map` in both places for `browserFlow`, thank you!

Reply:
Had to revert back to `mapNotNull` to fix:

`FAILURE TO REDEFINE Unable to perform redefinition of 'Lorg/mozilla/fenix/translations/TranslationsDialogBinding$onState$$inlined$mapNotNull$1;': Field 'this$0' (sig: Lorg/mozilla/fenix/translations/TranslationsDialogBinding;) is missing!`

(Trace reference to `mapNotNull` is a little suspect to me in the trace, but it might be happening in the later session combination. The issue stopped reproducing with testing once `map` -> `mapNotNull`.)

The error test behavior was not fetching the languages when a pre-translated state.

Issue only occurred  on an auto-translated page when restoring app from background during manual testing, moving back to `mapNotNull` fixed the issue.